src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -2060,12 +2269,43 @@ string ITupleInternal.ToStringEnd()
             }
         }
 
-        int ITupleInternal.Size
+        /// <summary>
+        /// The number of positions in this data structure.
+        /// </summary>
+        int ITuple.Length
+        {
+            get
+            {
+                return 7 + ((ITuple)Rest).Length;
```

Comment:
Casting to `ITuple` is going to box `Rest`. The box allocation can be avoided by doing:

```c#
return 7 + Rest.Length;
```